// react-loading-io/dist/esm/Spin/Spin.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import { clsx } from '../helpers';
import './spin.css';

var Spin = function Spin(_ref) {
  var className = _ref.className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? '#ED4585' : _ref$color,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 200 : _ref$size,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 10 : _ref$width,
      _ref$speed = _ref.speed,
      speed = _ref$speed === void 0 ? 1 : _ref$speed,
      style = _ref.style;
  var scale = size / 200;
  var translate = size * scale;
  var border = width * 4;
  return React.createElement("div", {
    className: clsx('rl-spin', className),
    style: _objectSpread({
      '--rl-spin-size': "".concat(size, "px"),
      '--rl-spin-color': color,
      '--rl-spin-scale': scale,
      '--rl-spin-translate': "".concat(translate, "px"),
      '--rl-spin-speed': "".concat(speed, "s"),
      '--rl-spin-border': "".concat(border, "px")
    }, style)
  }, React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)), React.createElement("div", null, React.createElement("div", null)));
};

process.env.NODE_ENV !== "production" ? Spin.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.number,
  width: PropTypes.number,
  style: PropTypes.object
} : void 0;
export default Spin;
